pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ConsumedLedgersTrimTest.java

```diff
@@ -142,6 +143,14 @@ public void testConsumedLedgersTrimNoSubscriptions() throws Exception {
         // the lastMessageId is still on the previous ledger
         restartBroker();
         // force load topic
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).until(() -> {
+            try {
+                pulsar.getBrokerService().getTopicIfExists(topicName).get(3, TimeUnit.SECONDS).get();
+                return true;
+            } catch (Exception ignored) {
+            }
```

Comment:
When a code block should pass, using `untilAsserted` will lead to cleaner code and there won't be a need to return a boolean or catch exceptions.

Reply:
The exception will not be automatically caught, you can try to throw a RuntimeException directly inside.
The test will be interrupted

Reply:
Yes you are right @315157973 , untilAsserted will only catch AssertionErrors .  I'm sorry about the incorrect advice.
Your original approach seemed to be the best approach after all since it would also handle the case of an exception being caught.

Reply:
hmm. there's also `.catchUncaughtExceptions()`.

Reply:
`catchUncaughtExceptions` defaults to true in Awaitility, so after all, untilAsserted should be fine without catching the exception. @315157973  Have you actually checked that RuntimeException terminates untilAsserted?